lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3400,7 +3400,15 @@ void SPIRVToLLVM::transIntelFPGADecorations(SPIRVValue *BV, Value *V) {
     SmallString<256> AnnotStr;
     generateIntelFPGAAnnotation(BV, AnnotStr);
     if (!AnnotStr.empty()) {
-      auto *GS = Builder.CreateGlobalStringPtr(AnnotStr);
+      Constant *GS = nullptr;
+      std::string StringAnnotStr = AnnotStr.c_str();
+      static unordered_map<std::string, Constant *> Annotations;
```

Comment:
In general `static` variables are not appreciated (for example in LLVM coding guidelines https://llvm.org/docs/CodingStandards.html#do-not-use-static-constructors ), because order of their initialization is not defined by C++ standard. Can we for example make this map a field of `SPIRVReader`?

Reply:
Check new commit please, This is what you're talking about?

Reply:
Yup